lib/aggregation/reporting/src/test/reporting-test.js

```diff
@@ -151,81 +151,110 @@ describe('abacus-usage-report', () => {
+
         request.get(
-          'http://localhost::p/v1/metering/organizations/' +
-          ':organization_id/aggregated/usage/:time', {
+          url, {
             p: server.address().port,
             organization_id: oid,
-            time: 1420574400000
+            time: time
           }, (err, val) => {
             expect(err).to.equal(undefined);
             expect(val.statusCode).to.equal(200);
             expect(val.body).to.deep.equal(expected);
-            done();
+            cb();
           });
       };
 
-      verify(false, () => verify(true, done));
+      before(() => {
+        const planAReport = planReportTemplate('basic',
+          builder.buildAggregatedUsage(1,
+          100, 300, {
+            consumed: 475200000,
+            consuming: 6
+          }, {
+            consumed: 10843200000,
+            consuming: 6
+          }, 1, 3, 45, { price: 0.00014 }, 114, 0.01596, true, true, true),
+          builder.buildWindow(undefined, undefined, undefined,
+          undefined, undefined, 49.01596));
+        const planBReport = planReportTemplate('standard',
+          builder.buildAggregatedUsage(
+          20, 200, 3000, {
+            consumed: 633600000,
+            consuming: 8
+          }, {
+            consumed: 14457600000,
+            consuming: 8
+          }, 10, 8, 540, { price: 0.00028 }, 152, 0.04256, true, true, true),
+          builder.buildWindow(undefined, undefined, undefined, undefined,
+            undefined, 558.04256));
+
+        const consumer1 = consumerReportTemplate('basic',
+          builder.buildAggregatedUsage(
+          undefined, undefined, undefined, undefined,
+          undefined,
+          1, 3, 45, undefined, undefined, 0.01596, undefined, undefined,
+          true), [planAReport], builder.buildWindow(undefined,
+          undefined, undefined, undefined, undefined, 49.01596));
+        const consumer2 = consumerReportTemplate('standard',
+        builder.buildAggregatedUsage(
+          undefined, undefined, undefined, undefined, undefined,
+          10, 8, 540, undefined, undefined, 0.04256, undefined, undefined,
+          true), [planBReport], builder.buildWindow(undefined, undefined,
+            undefined, undefined, undefined, 558.04256));
+
+        const id = 'k/a3d7fe4d-3cb1-4cc3-a831-ffe98e20cf27/t/0001420502400000';
+
+        expected = reportTemplate(id,
+          builder.buildWindow(undefined, undefined,
+          undefined, undefined, undefined, 607.05852),
+          builder.buildAggregatedUsage(undefined, undefined, undefined,
+            undefined, undefined, 11, 11, 585, undefined, undefined, 0.05852,
+            undefined, undefined,
+          true), [planAReport, planBReport], [consumer1, consumer2]);
+      });
+
+      it('successfully when throttle limit is not exceeded', (done) => {
+
+        verify(false, () => verify(true, done));
+      });
+
+      // Test would fail (time-out) when retrieveUsage doesn't have a callback
```

Comment:
We can remove this comment as this is the test used to verify the addition of the callback anyway